File: mpp-proiect-repository-cristinahendre/lab12/lab6/modul/DomainModule/src/main/java/Domain/Validators/DonatorValidator.java
package Domain.Validators;

import Domain.Donator;

public class DonatorValidator implements Validator<Donator> {

    /**
     * Valideaza un donator
     * @param entity ->donatorul
     * @throws ValidationException
     */
    @Override
    public void validate(Donator entity) throws ValidationException {
        if(entity.getNume().equals(" ") || entity.getNume().equals(""))
            throw new ValidationException("Numele este vid.");
        if(entity.getPrenume().equals(" ") || entity.getPrenume().equals(""))
            throw new ValidationException("Prenumele este vid.");
        if(entity.getAdresa().equals(" ") || entity.getAdresa().equals(""))
            throw new ValidationException("Adresa este vida.");
        if(entity.getNrTelefon() < 0)
            throw new ValidationException("Numarul de telefon e negativ.");


    }
}

File: mpp-proiect-repository-cristinahendre/C#/Cerinta1/Cerinta1/Domain/Validator/DonatorValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Cerinta1.Domain.Validator
{
    class DonatorValidator : IValidator<Donator>
    {

     /*
     * Valideaza un donator
     * @param entity ->donatorul
     * @throws ValidationException
     */

        public void Validate(Donator entity)
        {
            if (entity.Nume.Equals(" ") || entity.Nume.Equals(""))
                throw new ValidationException("Numele este vid");
            if (entity.Adresa.Equals(" ") || entity.Adresa.Equals(""))
                throw new ValidationException("Adresa este vida");
            if (entity.Prenume.Equals(" ") || entity.Prenume.Equals(""))
                throw new ValidationException("Prenumele este vid");
            if (entity.NrTelefon < 0)
                throw new ValidationException("Numarul de telefon  e negativ.");
        }
    }
}
